### PR TITLE
feat: Qwen2 support

### PR DESCRIPTION
**Pull request by maiqingqiang**
Add support for Qwen2

https://huggingface.co/Qwen

image

### CODE REVIEW

File: Tools/llm-tool/LLMTool.swift [hunk 1]
```diff
@@ -131,6 +133,8 @@ struct AsyncGenerator: AsyncParsableCommand {
 
         let modelConfiguration = ModelConfiguration.configuration(id: model)
         let (model, tokenizer) = try await load(configuration: modelConfiguration)
+        
+        print("Model loaded -> \(self.model)")
```
Comment (davidkoski) [inline Tools/llm-tool/LLMTool.swift:137]: These look like they might be for debugging -- do you mean to leave them in here?

File: Libraries/LLM/Qwen2.swift [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+        _lmHead.wrappedValue = Linear(args.hiddenSize, args.vocabularySize, bias: false)
+    }
+
+    public func callAsFunction(_ inputs: MLXArray, cache: [(MLXArray, MLXArray)]?) -> (
+        MLXArray, [(MLXArray, MLXArray)])
+    {
+        let (out, cache) = model(inputs, cache: cache)
+        return (lmHead(out), cache)
+    }
+}
```
Comment (davidkoski) [inline Libraries/LLM/Qwen2.swift:207]: Looks great!

File: Tools/llm-tool/LLMTool.swift (hunk 1 above)
Comment (maiqingqiang) [reply Tools/llm-tool/LLMTool.swift:137]: Yes. There are many models configured in Scheme. Sometimes I forget which model is currently being used, so I want to display the currently loaded model.

image

**Review by davidkoski — APPROVED** [review]
Thank you for the contribution!

**Comment by maiqingqiang** [comment]
@davidkoski I have a suggestion to separate Libraries from mlx-swift-examples, make it an independent package or put it in mlx-swift for easier reference by other projects. Similar to https://github.com/huggingface/transformers/tree/main/src/transformers/models

**Comment by davidkoski** [comment]
> @davidkoski I have a suggestion to separate Libraries from mlx-swift-examples, make it an independent package or put it in mlx-swift for easier reference by other projects. Similar to https://github.com/huggingface/transformers/tree/main/src/transformers/models

Currently it is set up a bit like the python code: 

- https://github.com/ml-explore/mlx-examples/tree/main/llms/mlx_lm

I will pass the request along and we will discuss.  Thank you!

**Comment by maiqingqiang** [comment]
> > @davidkoski I have a suggestion to separate Libraries from mlx-swift-examples, make it an independent package or put it in mlx-swift for easier reference by other projects. Similar to https://github.com/huggingface/transformers/tree/main/src/transformers/models
> 
> Currently it is set up a bit like the python code:
> 
> * https://github.com/ml-explore/mlx-examples/tree/main/llms/mlx_lm
> 
> I will pass the request along and we will discuss. Thank you!

@davidkoski @awni Hello, may I ask if there are any discussion results?

**Comment by johnmai-dev** [comment]
> @davidkoski I have a suggestion to separate Libraries from mlx-swift-examples, make it an independent package or put it in mlx-swift for easier reference by other projects. Similar to https://github.com/huggingface/transformers/tree/main/src/transformers/models

Hello @davidkoski @awni 

Is there a plan for Swift to also separate into packages now that Python has already done so?
https://github.com/ml-explore/mlx-lm

**Comment by ronaldmannak** [comment]
I also think separate packages make sense. Maybe that effort can be combined with improving the API of the libraries to make them more "Swift-like". I believe that would make it easier for new developers to start with MLX. I think @rudrankriyam mentioned some ideas and inspiration about the API on Twitter recently

**Comment by awni** [comment]
> Maybe that effort can be combined with improving the API of the libraries to make them more "Swift-like". I believe that would make it easier for new developers to start with MLX.

 What do you have in mind there? @ronaldmannak @rudrankriyam 

Would be really useful to hear your perspective on which parts of the API are not "Swift-like" and/or which parts could be easier to use.

**Comment by johnmai-dev** [comment]
> We haven't made a plan for it either way yet. Is it something that would be useful for you?

@awni, thank you for your reply!

Personally, I think using the name "examples" might mislead people into thinking it is just sample code rather than a formal and reusable library. If we split Libraries from mlx-swift-examples and make it an independent, modular, single-responsibility package, it would more clearly indicate its purpose and stability.

**Pull request merged**